core/domain/html_cleaner_test.py

```diff
@@ -360,3 +360,65 @@ def test_validate_text_angular(self):
         self.assertEqual(
             actual_output_without_migration,
             expected_output_without_migration)
+
+    def test_validate_soup_text_angular(self):
+        test_cases = [
+            (
+                '<p>Hello <b>this </b>is </p><p><br></p><p>test <b>case '
+                '</b>for </p><p><oppia-noninteractive-collapsible '
+                'content-with-value=\"&amp;quot;Hello oppia&amp;quot;\" '
+                'heading-with-value=\"&amp;quot;Learn more about APIs&'
+                'amp;quot;\"></oppia-noninteractive-collapsible><br></p><p>'
+                'for migration testing</p>'
+            ),
+            (
+                'Hello<div>oppia</div>testing <i>in progess</i>!'
+            ),
+            (
+                '<p>Hello</p><p>oppia</p><p>testing <i>in progress</i>!</p>'
+            )
+        ]
+
+        expected_output = [False, True, False]
+        err_dict = {}
+
+        for index, test_case in enumerate(test_cases):
+            actual_output = (
+                html_cleaner._validate_soup_for_textangular( # pylint: disable=protected-access
+                    bs4.BeautifulSoup(test_case, 'html.parser'),
+                    err_dict))
+
+            self.assertEqual(actual_output, expected_output[index])
+
+    def test_convert_tag_contents_to_text_angular(self):
+        test_cases = [{
+            'html_content': (
+                '<div>Hello <b>this </b>is </div><p><br></p><p>test <b>case '
```

Comment:
Apologies for missing this earlier, but can you add a similar test case for tabs?

Reply:
No problem :). Added the tests!